doc/contributing/writing-tests.md

```diff
@@ -141,6 +141,27 @@ request. Interesting things to notice:
 
 ## General recommendations
 
+### Usage of `node:test`
+
+It is optional to use `node:test` in tests outside of testing the `node:test`
```

Comment:
I think we should mention that this is for newly added tests, to avoid encouraging people update existing tests solely for stylistic preferences. If people have to update it to use `node:test` based on stylistic preferences, IMO we should require them to break it deliberately in the PR and and run the CI both before and after the update, and see what the output looks like, to see if the use of `node:test` makes the output look worse than before.